src/main/java/com/iexec/core/replicate/ReplicateSupplyService.java

```diff
@@ -132,7 +136,7 @@ public ReplicateSupplyService(ReplicatesService replicatesService,
             // no need to ge further if the consensus is already reached on-chain
             // the task should be updated since the consensus is reached but it is still in RUNNING status
             if (taskService.isConsensusReached(task)) {
```

Comment:
Why keeping isConsensusReached in TaskService? Should a now pure TaskService depends on ReplicateService?

Reply:
Not sure about this one. I was thinking about putting only transitional methods into `TaskUpdateManager`. `isConsensusReached` is still a Task-related methods but does not make a task transition from one state to another so I guessed it was still belonging to `TaskService`.
Do you think it should be in another service?

Reply:
2 points:
A - Why having this pre-transition check in a method (in a dedicated service) and still having other pre-transition checks with booleans without method? (non consistency)
B - This method doesn't need an access to the TaskRepo while other methods do (static-like)

Reply:
A - It's also used in `ReplicateSupplyService` which is different from other pre-transition checks booleans.
B - You're right.

I see your point and it's probably better for it to be in `TaskUpdateManager`. I'll move it. Thanks!

Reply:
OK for ReplicateSupplyService thanks.
We might see if a separate checker would make sense